storage/operation/guarantees.go

```diff
@@ -9,7 +10,15 @@ import (
 	"github.com/onflow/flow-go/storage"
 )
 
-// UnsafeInsertGuarantee inserts a [flow.CollectionGuarantee] into the database, keyed by the collection ID.
+// InsertGuarantee inserts a collection guarantee by ID.
+// Error returns:
+//   - storage.ErrAlreadyExists if the key already exists in the database.
+//   - generic error in case of unexpected failure from the database layer or encoding failure.
+func InsertGuarantee(w storage.Writer, guaranteeID flow.Identifier, guarantee *flow.CollectionGuarantee) error {
+	return UpsertByKey(w, MakePrefix(codeGuarantee, guaranteeID), guarantee)
```

Comment:
:warning: As far as I can see there is no `insert` anymore. I am a bit worried that some unhandled edge cases might happen because of this. Previously if we expect that there is no data and we use insert and suddenly `insert` returns an error it means we have encountered a bug in the logic. With new approach we use upsert everywhere which leads to overwriting values when we haven't explicitly asked about. I think this behavior is dangerous.

Reply:
Sorry, the comments was wrong. InsertGuarantee does not return any error, but IndexGuarantee does return `storage.ErrAlreadyExists` error. Updated in c9d9c16781